### PR TITLE
[skip ci]Cache on-imagebuilder debian packages

### PR DESCRIPTION
**Pull request by PengTian0**
For story [RAC-4260](https://rackhd.atlassian.net/browse/RAC-4260)

image

@panpan0000 @changev @yyscamper @anhou @keedya

### CODE REVIEW

File: build-release-tools/application/make_debian_packages.py [hunk 1]
```diff
@@ -230,20 +247,45 @@ def write_downstream_parameter_file(build_directory, manifest_file, is_official_
...
+    # exists in bintray
+    def packages_exist_filter(repo):
+        try:
+            repo_dir = os.path.join(build_directory, repo)
+            version_generator = VersionGenerator(repo_dir)
+            version = version_generator.generate_package_version(is_official_release)
+            if version is None:
+                return True
+            version_object = bintray.get_package_version_object(repo, version)
+            if version_object is not None:
```
Comment (anhou) [inline build-release-tools/application/make_debian_packages.py]: double 'no' 
'if version_object'

File: build-release-tools/application/make_debian_packages.py [hunk 2]
```diff
@@ -230,20 +247,45 @@ def write_downstream_parameter_file(build_directory, manifest_file, is_official_
...
         else:
             raise RuntimeError("commit-id of RackHD is None. Please check the manifest {0}".format(manifest_file))
-            
+        
         # Write downstream parameters to downstream parameter file.
         common.write_parameters(parameter_file, params)
     except Exception, e:
         raise RuntimeError("Failed to write downstream parameter file \ndue to {0}".format(e))
 
+def make_packages_filter(bintray, build_directory, is_official_release):
```
Comment (anhou) [inline build-release-tools/application/make_debian_packages.py]: from the function name, it seems it has 'make' action in this function, actually it doesn't

File: build-release-tools/application/make_debian_packages.py [hunk 3]
```diff
@@ -230,20 +247,45 @@ def write_downstream_parameter_file(build_directory, manifest_file, is_official_
...
+    # The filter will return False if the version of the repository 
+    # exists in bintray
```
Comment (anhou) [inline build-release-tools/application/make_debian_packages.py]: why 'version' exist but return false, not true? be confused by the function name and the function annotation. please refactor this function name and annotation.

File: jobs/build_docker/build_docker.sh [hunk 4]
```diff
@@ -1,41 +1,45 @@
-#!/bin/bash
+#!/bin/bash -ex
```
Comment (panpan0000) [inline jobs/build_docker/build_docker.sh]: are you sure with set -x ? will it expose $SUDO_PASSWORD ?

File: jobs/create_manifest/create_manifest.sh [hunk 5]
```diff
@@ -24,8 +24,8 @@ manifest_name=$(find -maxdepth 1 -name "$slicedBranch-[0-9]*" -printf "%f\n")
 
 MANIFEST_FILE_URL=https://dl.bintray.com/$BINTRAY_SUBJECT/$BINTRAY_REPO/$manifest_name
 echo "MANIFEST_FILE_URL=<a href=\"$MANIFEST_FILE_URL\">$manifest_name</a>"
-echo "MANIFEST_FILE_URL=$MANIFEST_FILE_URL" > downstream_file
-
+echo "MANIFEST_FILE_URL=$MANIFEST_FILE_URL" >> downstream_file
```
Comment (panpan0000) [inline jobs/create_manifest/create_manifest.sh:27]: where was downstream file born ?

File: jobs/build_docker/build_docker.sh [hunk 6]
```diff
@@ -1,41 +1,45 @@
...
+--is-official-release $IS_OFFICIAL_RELEASE )
 
-echo $SUDO_PASSWORD |sudo -S ./build_all.sh
+common_path=/var/renasar/on-http/static/http/common
+pxe_path=/var/renasar/on-tftp/static/tftp
 
-#copy to on-imagebuilder folder for docker build
-output_path=/tmp/on-imagebuilder
+echo $SUDO_PASSWORD |sudo -S rm -rf /etc/apt/sources.list.d/rackhd.source.list
+echo "deb https://dl.bintray.com/rackhd/debian trusty main" | sudo tee -a /etc/apt/sources.list.d/rackhd.source.list
```
Comment (panpan0000) [inline jobs/build_docker/build_docker.sh]: I think this should be rackhd-mirror, this is the cached version you need. build from previous "debian build" step.

File: jobs/build_docker/build_docker.sh (hunk 4 above)
Comment (PengTian0) [reply jobs/build_docker/build_docker.sh]: Thanks!
The plugin will mark the password.
But I'll just remove it.

File: jobs/create_manifest/create_manifest.sh (hunk 5 above)
Comment (PengTian0) [reply jobs/create_manifest/create_manifest.sh:27]: It's created by the command ```echo "MANIFEST_FILE_URL=$MANIFEST_FILE_URL" >> downstream_file```

**Review by anhou — APPROVED** [review]
👍 with commits are squashed

**Comment by panpan0000** [comment]
👍   looks good now .

**Pull request merged**